analyses/bangladesh/trigger_development/read_in_data.py

```diff
@@ -0,0 +1,173 @@
+from src.bangladesh import get_glofas_data as ggd
+
+DATA_DIR = Path(os.environ["AA_DATA_DIR"])
+GLOFAS_DIR = DATA_DIR / "processed/bangladesh/GLOFAS_Data"
+STATION = "Bahadurabad_glofas"
+ffwc_dir = DATA_DIR / 'exploration/bangladesh/FFWC_Data'
+
+
+def get_glofas_reanalysis():
+    glofas_reanalysis = glofas.GlofasReanalysis(stations_lon_lat=ggd.FFWC_STATIONS)
+    da_glofas_reanalysis = glofas_reanalysis.read_processed_dataset(
+        country_name=ggd.COUNTRY_NAME, country_iso3=ggd.COUNTRY_ISO3
+    )[STATION]
+    return da_glofas_reanalysis
+
+
+def get_glofas_forecast():
+    glofas_forecast = glofas.GlofasForecast(
+        stations_lon_lat=ggd.FFWC_STATIONS, leadtime_hours=ggd.LEADTIME_HOURS
+    )
+    da_glofas_forecast_dict = {
+        leadtime_hour: glofas_forecast.read_processed_dataset(
+            country_name=ggd.COUNTRY_NAME,
+            country_iso3=ggd.COUNTRY_ISO3,
+            leadtime_hour=leadtime_hour,
+        )[STATION]
+        for leadtime_hour in ggd.LEADTIME_HOURS
+    }
+    da_glofas_forecast_dict = shift_dates(da_glofas_forecast_dict)
+    return convert_dict_to_da(da_glofas_forecast_dict)
+
+
+def get_glofas_reforecast():
+    glofas_reforecast = glofas.GlofasReforecast(
+        stations_lon_lat=ggd.FFWC_STATIONS, leadtime_hours=ggd.LEADTIME_HOURS
+    )
+    da_glofas_reforecast_dict = {
+        leadtime_hour: glofas_reforecast.read_processed_dataset(
+            country_name=ggd.COUNTRY_NAME,
+            country_iso3=ggd.COUNTRY_ISO3,
+            leadtime_hour=leadtime_hour,
+        )[STATION]
+        for leadtime_hour in ggd.LEADTIME_HOURS
+    }
+    da_glofas_reforecast_dict = interp_dates(shift_dates(da_glofas_reforecast_dict))
+    return convert_dict_to_da(da_glofas_reforecast_dict)
+
+
+def shift_dates(da_dict):
+    return {
+        leadtime_hour: da.assign_coords(
+            time=da.time.values + np.timedelta64(int(leadtime_hour / 24), "D")
+        )
+        for leadtime_hour, da in da_dict.items()
+    }
+
+
+def interp_dates(da_dict):
+    return {
+        leadtime_hour: da.interp(
+            time=pd.date_range(da.time.min().values, da.time.max().values),
+            method="linear",
+        )
+        for leadtime_hour, da in da_dict.items()
+    }
+
+
+def convert_dict_to_da(da_glofas_dict):
+    # Create time array that accounts for all the shifts
+    time = np.arange(
+        da_glofas_dict[min(da_glofas_dict.keys())].time.values[0],
+        da_glofas_dict[max(da_glofas_dict.keys())].time.values[-1]
+        + np.timedelta64(1, "D"),
+        dtype="datetime64[D]",
+    )
+    da_glofas_dict = {
+        leadtime_hour: da_glofas.reindex({"time": time})
+        for leadtime_hour, da_glofas in da_glofas_dict.items()
+    }
+
+    data = np.array([da_glofas.values for da_glofas in da_glofas_dict.values()])
+    return xr.DataArray(
+        data=data,
+        dims=["leadtime_hour", "number", "time"],
+        coords=dict(
+            number=da_glofas_dict[120].number,
```

Comment:
I'm a bit unclear on what ```number``` is here...

Reply:
Oops good point, it's the ensemble member number, I will clarify this